app_dart/lib/src/service/branch_service.dart

```diff
@@ -30,6 +34,44 @@ class BranchService {
   final GerritService gerritService;
   final RetryOptions retryOptions;
 
+  /// Add a [CreateEvent] branch to Datastore.
+  Future<void> handleCreateRequest(CreateEvent createEvent) async {
+    log.info('the branch parsed from string request is ${createEvent.ref}');
+
+    final String? refType = createEvent.refType;
+    if (refType == 'tag') {
+      log.info('create branch event was rejected because it is a tag');
+      return;
+    }
+    final String? branch = createEvent.ref;
+    if (branch == null) {
+      log.warning('Branch is null, exiting early');
+      return;
+    }
+    final String repository = createEvent.repository!.slug().fullName;
+    final int lastActivity = createEvent.repository!.pushedAt!.millisecondsSinceEpoch;
+    final bool forked = createEvent.repository!.isFork;
+
+    if (forked) {
+      log.info('create branch event was rejected because the branch is a fork');
+      return;
+    }
+
+    final String id = '$repository/$branch';
+    log.info('the id used to create branch key was $id');
+    final DatastoreService datastore = DatastoreService.defaultProvider(config.db);
+    final Key<String> key = datastore.db.emptyKey.append<String>(Branch, id: id);
+    final Branch currentBranch = Branch(key: key, lastActivity: lastActivity);
+    try {
+      await datastore.lookupByValue<Branch>(currentBranch.key);
+    } on KeyNotFoundException {
+      log.info('create branch event was successful since the key is unique');
+      await datastore.insert(<Branch>[currentBranch]);
+    } catch (e) {
```

Comment:
If this isn't catching a specific exception, we should remove the catch clause

https://dart.dev/effective-dart/usage#avoid-catches-without-on-clauses

Reply:
Umm removed. 

Umm i thought last time in standup Casey mentioned that we do not want to break the workflow of the request. Umm I was thinking if we don't catch other errors, this would cause the webhook request to broke. But maybe we are fine with webhook request for branch to be broke, since it wouldn't affect infra workflows?

Reply:
Yes, breaking on the release branch requests is expected